### ISSUE TITLE
python-forge, markdown-it-py dependencies are still mandatory

### ISSUE THREAD

**Issue by layday** [requests-cache/aiohttp-client-cache#175]
python-forge and markdown-it-py are optional in Poetry's configuration but are not associated with an extra.  The only way to mark a dependency as being optional in [core metadata](https://packaging.python.org/en/latest/specifications/core-metadata/) is by association with an extra, e.g. `Requires-Dist: foo ; extra == "bar"`. The Poetry builder does not omit or raise for unlinked optional dependencies, but emits them without an `extra` marker, meaning that they are installed by default.

**Comment by JWCook** [comment]
Oh, I wasn't aware of that behavior. Thanks for letting me know.

**Comment by JWCook** [comment]
Fixed in v0.9.1. I verified that only the expected dependencies are installed now:
```py
aiohttp-client-cache==0.9.1
    itsdangerous==2.1.2
    url-normalize==1.4.3
        six==1.16.0
    aiohttp==3.8.5
        aiosignal==1.3.1
        async-timeout==4.0.3
        attrs==23.1.0
        charset-normalizer==3.2.0
        frozenlist==1.4.0
        idna==3.4
        multidict==6.0.4
        yarl==1.9.2
```

**Comment by layday** [comment]
Thanks!
